### DISCUSSION TITLE
Flashing the sense-u 2 hub

### DISCUSSION THREAD

**Discussion by lordratner** [andyboeh/esphome-sense-u#7] [Q&A] [answered]
Did you have to open it up, or were you able to do it entirely with the externally accessible USB port?

**Comment by andyboeh** [comment]
I opened mine, otherwise I wouldn't have been able to find out about the Rx/Tx/IO0 pins on the USB port ;)

**Reply by lordratner** [reply]
> I opened mine, otherwise I wouldn't have been able to find out about the Rx/Tx/IO0 pins on the USB port ;)

Ok cool. Strangely though I can't get windows to recognize the device when I plug it in, and I've installed all of the referenced drivers in the ESPHome connection prompt. Did it show up the same way as a regular ESP32 dev board (or similar)?

**Reply by andyboeh** [reply]
NEVER connect the device via USB to a computer - I wrote that in bold into the README, twice! Rx, Tx and IO0 are not USB signals, they are TTL UART signals. They did not include a USB <-> UART bridge into the PCB, so you need an external one.

**Reply by lordratner** [reply]
Lol, sorry, I didn't even see you had redone the readme. I'll dig through that first🤦‍♂️.

**Reply by lordratner** [reply]
Got a recommendation for a USB-ttl cable? I haven't added that to my kit yet

**Reply by andyboeh** [reply]
I own several different ones and I have good experience with CP2102- or FT232H-based ones. I flashed this one with an Adafruit FT232H breakout board. Just make sure it uses 3.3V signal levels (which seems to be the default nowadays).

**Reply by lordratner** [reply]
I think this one is the same thing, I only need the 3.3, gnd, tx, and rx pins, correct?

https://smile.amazon.com/dp/B09FLH43TS?psc=1&smid=A2SA28G0M1VPHD&ref_=chk_typ_imgToDp

I've never used one of these things, are they just plug and play, or do I need to load something onto it to work with the ESP32?

Thanks for your help, hopefully I can contribute more once I get it flashed.

**Reply by andyboeh** [reply]
Right, that should work. Just connect ESP32_Tx <-> Rx, ESP32_Rx <-> Tx, IO0 <-> GND and GND <-> GND. Do not connect USB to a computer, use a dedicated power supply instead or power the ESP32 directly by 3.3V or 5V.

**Comment by lordratner** [comment, marked as answer]
IO0 and GND to Ground?

Then power independently? That board is able to send out either 3.3 or 5v, is it ok to power the Sense-u base from that?

So USB from computer to FT232 board. Then from the FT board to the Sense-u:

FT Rx --> ESP Tx
FT Tx --> ESP Rx
FT Gnd --> ESP Gnd
FT Gnd --> ESP GPIO0 (this allows the flashing to work, correct?)
FT VCCIO or USB Power Supply --> ESP 5v or 3.3v, whichever matches the supply

Then the computer will just recognize the ESP32?

**Reply by andyboeh** [reply]
Exactly, power it either from 5V or 3.3V, just don't apply 5V to the 3.3V input ;)

Correct, IO0 needs to be grounded to enter flash mode. Only one red LED will light up on the base station if that's correctly applied.

The computer will recognize a serial port (your USB <-> UART converter) and using this port esptool.py or any other ESP flasher will detect the ESP32.

**Reply by lordratner** [reply]
That worked like a treat, thanks!
